qt-bugs@ issue : not yet
Trolltech task ID : not yet
bugs.kde.org number : none, found by SaroEngels
applied: no
author: Christian Ehrlicher <ch.ehrlicher@gmx.de>

This small piece of code crashes Qt on windows (temp.jpg must be 32bpp):
--------------------8<-------------
QPixmap *backPixmap;
QPainter *mixedPainter;
{
  QImage backImage("D:\\temp.jpg");
  backPixmap = new QPixmap( QPixmap::fromImage( backImage ) );
  mixedPainter = new QPainter( backPixmap );
}
delete mixedPainter;
delete backPixmap;
--------------------8<-------------
QPixmap::detach() forgets to detach from the QImage. Later QPainter::begin()
saves a pointer to the paintEngine *before* the detach is done. So QPainter
paints on the wrong paintEngine and we get a warning + crash.


Index: src/gui/image/qpixmap.cpp
===================================================================
--- src/gui/image/qpixmap.cpp	(revision 813015)
+++ src/gui/image/qpixmap.cpp	(working copy)
@@ -1813,6 +1813,7 @@
     if (rasterData->texture)
         rasterData->texture->Release();
     rasterData->texture = 0;
+    rasterData->image.detach();
 #endif
 
     if (qt_pixmap_cleanup_hook_64 && data->ref == 1)
